lib/capybara/cuprite/browser/javascripts/index.js

```diff
@@ -139,20 +143,29 @@ class Cuprite {
       let min = ("0" + value.getMinutes()).slice(-2);
       let sec = ("0" + value.getSeconds()).slice(-2);
       this.setValue(node, `${year}-${month}-${date}T${hour}:${min}:${sec}`);
+      this.input(node);
     } else {
       for (let i = 0; i < value.length; i++) {
         let char = value[i];
         let keyCode = this.characterToKeyCode(char);
-        this.keyupdowned(node, "keydown", keyCode);
-        this.setValue(node, node.value + char);
+        // call the following functions in order, if one returns false (preventDefault),
+        // stop the call chain
+        [
+          () => this.keyupdowned(node, "keydown", keyCode),
+          () => this.keypressed(node, false, false, false, false, char.charCodeAt(0), char.charCodeAt(0)),
+          () => {
+            this.setValue(node, node.value + char)
+            this.input(node)
+          }
+        ].some(fn => fn())
```

Comment:
this is more readable than writing as a bunch of nested if statements, but at the cost of creating a few more memory objects per call (one array, three anonymous functions). Not sure if that matters.

Reply:
That's totally fine for me